tardis_kromer_plot.py

```diff
@@ -304,16 +304,105 @@ def line_in_L(self):
     @property
     def line_info(self):
         """produces list of elements to be included in the kromer plot"""
-        line_out_infos_within_xlims = self.line_out_infos.loc[
-            (self.line_out_infos.wavelength >= self._xlim[0])
-            & (self.line_out_infos.wavelength <= self._xlim[1])
+
+        # gets list of elements and number of emitted packets
+        self.last_line_interaction_out_id = self.line_out_infos
+        self.last_line_interaction_out_angstrom = self.line_out_nu.to(
+            units.Angstrom, equivalencies=units.spectral()
+        )
+
+        self.lines = self.mdl.lines
+
+        packet_out_filter = (
+            self.last_line_interaction_out_angstrom
+            > self._xlim[0] * units.angstrom
+        ) & (
+            self.last_line_interaction_out_angstrom
+            < self._xlim[1] * units.angstrom
+        )
+
+        self.last_line_interaction_out_id[
+            "emitted_wavelength"
+        ] = self.last_line_interaction_out_angstrom
+
+        line_out_infos_within_xlims = self.last_line_interaction_out_id.loc[
+            (
+                self.last_line_interaction_out_id.emitted_wavelength
+                >= self._xlim[0]
+            )
+            & (
+                self.last_line_interaction_out_id.emitted_wavelength
+                <= self._xlim[1]
+            )
         ]
-        self._elements_in_kromer_plot = np.c_[
+        self._elements_in_kromer_plot_out = np.c_[
             np.unique(
                 line_out_infos_within_xlims.atomic_number.values,
                 return_counts=True,
             )
         ]
+
+        # gets list of elements and number of absorbed packets
+        self.last_line_interaction_in_id = self.line_in_infos
+        self.last_line_interaction_in_angstrom = self.line_in_nu.to(
+            units.Angstrom, equivalencies=units.spectral()
+        )
+
+        self.lines = self.mdl.lines
+
+        packet_in_filter = (
```

Comment:
packet_in_filter doesn't seem to be used anywhere in the code. Is this filter necessary?

Reply:
correct - this part has been removed, along with the self.lines variable which is also unused. See #42